### PR TITLE
MCO: bump golang to 1.12

### PR DESCRIPTION
**Pull request by runcom**
Part of https://github.com/openshift/machine-config-operator/issues/796

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

File: ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.1.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ build_root:
     cluster: https://api.ci.openshift.org
     name: release
     namespace: openshift
-    tag: golang-1.10
+    tag: golang-1.12
```
Comment (cgwalters) [inline ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.1.yaml]: Should we really be changing the `4.1` file?

Comment (runcom) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.1.yaml]: Oh gosh great catch

Comment (runcom) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-release-4.1.yaml]: fixed now!

**Pull request merged**